File: go/maximum-absolute-sum-of-any-subarray.go
func maxAbsoluteSum(nums []int) int {
	currentSum := 0
	minimal := 0
	maximal := 0
	for i := 0; i < len(nums); i++ {
		currentSum += nums[i]
		if maximal < currentSum {
			maximal = currentSum
		}
		if currentSum < minimal {
			minimal = currentSum
		}
	}
	difference := maximal - minimal
	if difference < 0 {
		difference = -difference
	}
	return difference
}

File: python/maximum-absolute-sum-of-any-subarray.py
class Solution:
    def maxAbsoluteSum(self, nums: List[int]) -> int:
        current_sum = 0
        minimal = 0
        maximal = 0
        for num in nums:
            current_sum += num
            minimal = min(minimal, current_sum)
            maximal = max(maximal, current_sum)
        return abs(maximal - minimal)
